src/Altinn.App.Core/Internal/Process/ProcessTasks/Common/ProcessTaskFinalizer.cs

```diff
@@ -164,7 +164,9 @@ DataType dataType
             else
             {
                 // Remove the shadow fields from the data
-                data = JsonSerializer.Deserialize(serializedData, modelType)!;
+                data =
+                    JsonSerializer.Deserialize(serializedData, modelType)
+                    ?? throw new JsonException("Could not deserialize back datamodel after removing shadow fields");
```

Comment:
```suggestion
                    ?? throw new JsonException("Data was \"null\"");
```

Reply:
I still think the existing context is useful, but extended the message to include what you suggested